### ISSUE TITLE
is add_loss affected by sample weights? 

### ISSUE THREAD

**Issue by HansLeonardVanBrueggemann** [keras-team/keras#12337]
Dear all, 

for a certain network application, I have a KL loss computed among feature maps inside the network, and a loss function on the classification head. Now, for X samples I'm feeding during training, the KL loss must be computed and used for backprop **in all** X samples, whereas the classification loss is computed only **for some** of the X samples, thus setting the sample weights to 0. 

You can think of it as a VAE, where only for some of the samples in the batch I want the reconstruction loss to be backproped, but I want the KL loss to be backproped for all samples in the batch. 

Now, my question is: since I use model.add_loss to feed the KL loss back to the network, is this affected by the sample weighting on the classification head?
